tools/capture-test.py

```diff
@@ -71,6 +73,7 @@ def parse_opts():
     opts = ap.parse_args()
     if not opts.capchan:
         opts.capchan = opts.chan
+    opts.capmap = [int(x) for x in opts.capmap]
```

Comment:
```python
  assert len(opts.capmap) == opts.capchan
```

(never silently discard user input)

Reply:
I think you missed my point, which was only about "input validation". What happens if the user screws up and opt.capchan and opt.capmap are inconsistent with each other? More specifically, what happens to spurious/extra digits in `opts.capmap`? Whatever happens, it should never silently discard part of the user input. At the very least a `[log.]warning`. It's also a useful statement for the code reader (it would have helped me at least).